### PR TITLE
[NEAT-363] 🔥 Speedup tests

### PR DESCRIPTION
**Pull request by doctrino**
* Switch to oxy graph
* Some code speedups
* Deleted unused test data.

Overall 1m12s to 18-21s on my machine (only unit tests)

I skipped parallelization as 18s is so fast that the overhead of extra workers is unlikely to pay off.

### CODE REVIEW

File: tests/tests_unit/rules/test_importers/test_inference_importer.py [hunk 1]
```diff
@@ -8,13 +8,20 @@
...
 
     assert len(rules.properties) == 312
     assert len(rules.classes) == 59
 
     # checking multi-value type
-    assert set(rules.properties.data[19].value_type.types) == set(
+    multi_value_property = "OperatingShare.PowerSystemResource"
+    prop = next((prop for prop in rules.properties if prop.property_ == multi_value_property), None)
+    assert prop is not None, f"Failed to infer expected multi-value property {multi_value_property}"
+    assert set(prop.value_type.types) == set(
```
Comment (doctrino) [inline tests/tests_unit/rules/test_importers/test_inference_importer.py:24]: oxy seems to be not be deterministic, so need a most robust check here

File: cognite/neat/exceptions.py [hunk 2]
```diff
@@ -102,6 +102,9 @@ def __str__(self) -> str:
         return self.message
 
 
+class NeatValueError(NeatException, ValueError): ...
```
Comment (doctrino) [inline cognite/neat/exceptions.py:105]: It is good practice to add our own exception were it is a user mistake. Then we can later have for example sentry catch all non Neat error which would be bugs in the code.

File: cognite/neat/rules/importers/_spreadsheet2rules.py [hunk 3]
```diff
@@ -131,25 +131,24 @@ def sheet_names(self, role: RoleTypes) -> set[str]:
...
-                metadata = self.metadata
-            else:
-                metadata = self._read_metadata(excel_file, filepath)
-                if metadata is None:
-                    # The reading of metadata failed, so we can't continue
-                    return None
-
-            sheets, read_info_by_sheet = self._read_sheets(excel_file, metadata.role)
-            if sheets is None or self.issue_list.has_errors:
+    def read(self, excel_file: pd.ExcelFile, filepath: Path) -> None | ReadResult:
```
Comment (doctrino) [inline cognite/neat/rules/importers/_spreadsheet2rules.py:134]: Switch this one to opening the excel file outside. It is faster than opening the excel files for each set of sheets we want to read.

File: tests/tests_unit/test_neat_package_metadata.py [hunk 4]
```diff
@@ -35,6 +36,16 @@ def test_consistent_version_variables():
...
 
 def test_no_spaces_in_sub_folders() -> None:
-    name_by_location = {path: path.name for path in (ROOT / "cognite").rglob("*") if path.is_dir() and " " in path.name}
+    name_by_location: dict[Path, str] = {}
+    to_check = [ROOT / "cognite"]
+    while to_check:
+        current = to_check.pop()
+        for path in current.iterdir():
+            if path.is_dir():
+                if path.name not in {"node_modules"}:
```
Comment (doctrino) [inline tests/tests_unit/test_neat_package_metadata.py:45]: Avoid searching the `node_modules` shaves off 15 seconds on this test.

File: cognite/neat/graph/queries/_base.py [hunk 5]
```diff
@@ -98,55 +98,48 @@ def describe(
...
 
         Returns:
             Dictionary of instance properties
         """
-
         property_values: dict[str, list[str]] = defaultdict(list)
-
-        for subject, predicate, object_ in cast(list[ResultRow], self.graph.query(f"DESCRIBE <{instance_id}>")):
-            if object_.lower() not in [
+        identifier = remove_namespace_from_uri(instance_id, validation="prefix")
```
Comment (doctrino) [inline cognite/neat/graph/queries/_base.py:112]: The subject will be the same for each iteration, so only calling `remove_namespace_uri` once on it save some time

File: cognite/neat/utils/rdf_.py [hunk 6]
```diff
@@ -10,25 +12,28 @@
...
 
 def remove_namespace_from_uri(
-    *URI: URIRef | str | tuple[URIRef | str, ...],
+    URI: URIRef | str | SequenceNotStr[URIRef | str],
+    *,
     special_separator: str = "#_",
     validation: Literal["full", "prefix"] = "prefix",
-) -> tuple[str, ...] | str:
+) -> str | list[str]:
     """Removes namespace from URI
```
Comment (doctrino) [inline cognite/neat/utils/rdf_.py:37]: The overload was not correctly implemented leading to some unnecessary casting. So fixed the implementation

File: cognite/neat/graph/stores/_base.py [hunk 7]
```diff
@@ -221,7 +223,7 @@ def _parse_file(
         Args:
             filepath : File path to file containing graph data, by default None
             mime_type : MIME type of graph data, by default "application/rdf+xml"
```
Comment (nikokaoja) [inline cognite/neat/graph/stores/_base.py:225]: Revert to previous as base_uri is not always needed to load file to oxigraph. It is only needed when RDF file contains relative URIs, which is the case with Nordic44. In majority of cases there are absolute URI.

File: tests/tests_unit/rules/test_importers/test_inference_importer.py (hunk 1 above)
Comment (nikokaoja) [reply tests/tests_unit/rules/test_importers/test_inference_importer.py:24]: The order of processing ?

Comment (doctrino) [reply tests/tests_unit/rules/test_importers/test_inference_importer.py:24]: Running this tests a few times the "OperatingShare.PowerSystemResource" property is not always on index 19, it changes from run to run.

File: cognite/neat/graph/stores/_base.py (hunk 7 above)
Comment (doctrino) [reply cognite/neat/graph/stores/_base.py:225]: Got it, will revert this change.

File: tests/tests_unit/rules/test_importers/test_inference_importer.py (hunk 1 above)
Comment (nikokaoja) [reply tests/tests_unit/rules/test_importers/test_inference_importer.py:24]: that is expected, I think with in-memory rdflib store they do indexing of triples or something similar to maintain the order

**Review by nikokaoja — APPROVED** [review]

**Pull request merged**